### PR TITLE
Clean up build files after Ninja Builder is done

### PR DESCRIPTION
**Pull request by eliheuer**
This fixes issue https://github.com/googlefonts/gftools/issues/719

### CODE REVIEW

File: Lib/gftools/builder/_ninja.py [hunk 1]
```diff
@@ -61,6 +62,23 @@ def build(self):
...
+        # Clean up temp build files
+        search_directory = os.getcwd()
+        target_names = ["build.ninja", ".ninja_log", "instance_ufo", "master_ufo"]
+        for root, dirs, files in os.walk(search_directory, topdown=False):
+            for file in files:
+                if file in target_names:
+                    file_path = os.path.join(root, file)
+                    os.remove(file_path)
+                    print(f"Removed file: {file_path}")
+        for dir_name in dirs:
```
Comment (simoncozens) [inline Lib/gftools/builder/_ninja.py:74]: Shouldn't this be within the "os.walk" loop?

**Review by simoncozens — CHANGES_REQUESTED** [review]

File: Lib/gftools/builder/_ninja.py (hunk 1 above)
Comment (eliheuer) [reply Lib/gftools/builder/_ninja.py]: Sure, like this? I tested this with a few different font repos and it seems to work, and pushed it as an update to this PR (e558616).
```python
        # Clean up temp build files
        search_directory = os.getcwd()
        target_names = ["build.ninja", ".ninja_log", "instance_ufo", "master_ufo"]
        for root, dirs, files in os.walk(search_directory, topdown=False):
            for file in files:
                if file in target_names:
                    file_path = os.path.join(root, file)
                    os.remove(file_path)
                    print(f"Removed file: {file_path}")
            for dir_name in dirs[:]:
                if dir_name in target_names:
                    dir_path = os.path.join(root, dir_name)
                    shutil.rmtree(dir_path)
                    print(f"Removed directory: {dir_path}")
```

Comment (eliheuer) [reply Lib/gftools/builder/_ninja.py]: The shallow copy of dirs `dirs[:]` was ChatGPTs idea, but I looked into it and it made sense to me. ChatGPTs explanation:
>By creating a copy with dirs[:], you can safely remove elements from the original dirs list within the loop without affecting the ongoing iteration. This is particularly useful when you want to conditionally remove items from a list while iterating over it.

Comment (simoncozens) [reply Lib/gftools/builder/_ninja.py]: Nice, thanks.

**Pull request merged**